packages/flutter_tools/lib/src/project.dart

```diff
@@ -459,6 +460,13 @@ class AndroidProject extends FlutterProjectPlatform {
   /// Returns true if the current version of the Gradle plugin is supported.
   late final bool isSupportedVersion = _computeSupportedVersion();
 
+  Future<List<String>> getBuildOptions() async {
```

Comment:
I am also new to this @christopherfujino may know more. But this is what I assuming when I read the code.
AndroidProject class is responsible for all interactions from/to android sub project in the flutter folder. The vmservice is to register callback to the VMService so that whichever other dart running instances (in this case, Devtool) that connect to the VMService can call into the callbacks to get what it needs